src/components/lib/cos_kernel_api.c

```diff
@@ -476,7 +477,7 @@ cos_compinfo_alloc(struct cos_compinfo *ci, vaddr_t heap_ptr, vaddr_t entry,
 	compc = cos_comp_alloc(ci_resources, ctc, ptc, entry);
 	assert(compc);
 
-	cos_compinfo_init(ci, ptc, ctc, compc, heap_ptr, 0, ci_resources);
+	cos_compinfo_init(ci, ptc, ctc, ptc, compc, heap_ptr, 0, ci_resources);
```

Comment:
I'm starting to wonder if the right way to implement this is *not* to include a separate pgtbl in `cos_compinfo`, and rather to just set up a `cos_compinfo` that uses entirely global resources (the superblock should provide these, right?).  If we want a component to use local, we set the `cos_meminfo` to be the default `cos_compinfo`.  If we want a component to use global, we set `cos_meminfo` in it to the global `cos_compinfo`.

This would make it hard to use a combination of local and global, but that could be done using multiple `cos_compinfo`s to represent different portions of the memory and capability namespaces.